docs/DeveloperGuide.adoc

```diff
@@ -563,6 +585,71 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+[IMPORTANT]
+There are certain naming conventions which have to be followed on Windows operating system which are not necessary on Linux and MacOS operating systems. You can read more about the naming conventions for Windows link:https://docs.microsoft.com/en-us/windows/desktop/FileIO/naming-a-file#naming-conventions[here]. You can also refer to link:https://en.wikipedia.org/wiki/Filename#Comparison_of_filename_limitations[this] Wikipedia article for a detailed comparison between different naming conventions for different operating systems.
+* Ensure consistency by creating a file naming convention.
+* Ensure that files created with file names specified by the user can be used on Windows, Linux and MacOS operating systems. This is to avoid potential bugs involving file names as MediTabs supports Windows, Linux and MacOS operating systems hence, the user might switch between these operating systems when using MediTabs.
+
+==== Current Implementation
+
+The validation of file name mechanism is facilitated mainly by the `FileName#isValidFileName(String fileNameToCheck)` operation. The operation checks if the `fileNameToCheck` input is a valid file name based on the file naming convention we have created for MediTabs.
+
+The file naming convention used in MediTabs is as follows: +
+Format: `Start with an alphabet or number followed by alphabets, numbers, underscore or hyphen`
+
+[NOTE]
+In order to ensure that the validation of file name is platform independent, the validation is implemented such that it ensures that the specified file name not only follows the file naming convention used in MediTabs but also does not violate file naming conventions in any platform especially Windows which has a very strict file name conventions.
+
+[IMPORTANT]
+Based on manual testing by our team on Windows, it is discovered that `COM0` and `LPT0` are also part of the reserved names which are not allowed as file names in Windows. However, they are not reflected in the Windows documentation under the link:https://docs.microsoft.com/en-us/windows/desktop/FileIO/naming-a-file#naming-conventions[naming conventions] section. In order to ensure that `FileName` class validation implementation is platform independent, we have included `COM0` and `LPT0` as reserved names when validating the specified file name.
+
+A full list of reserved names implemented in `FileName` class which are not allowed as file names is as follows:
+
+* `CON, PRN, AUX, NUL, COM0, COM1, COM2, COM3, COM4, COM5, COM6, COM7, COM8, COM9, LPT0, LPT1, LPT2, LPT3, LPT4, LPT5, LPT6, LPT7, LPT8, and LPT9`
+
+[WARNING]
+The specified file name *must not* be an empty string or `null`. If the specified file name is an empty string, the specified file name will be declared as invalid. If the specified file name is `null`, a `NullPointerException` will be thrown.
+
+==== How to integrate it into your project or code?
+
```

Comment:
Not sure if this section is appropriate since it is not about MediTabs

Reply:
It is part of MediTabs when they improve the MediTabs project this teaches them how to integrate it to their own MediTabs project etc. Furthermore, the target audience is developers for developers guide so developers might find it useful.